yang/frr-bgp-common.yang

```diff
@@ -0,0 +1,1092 @@
+         reset session if a link to a directly connected external peer goes down.";
+    }
+
+    leaf local-pref {
+      type uint32;
+      default "100";
+      description
+        "BGP local preference attribute sent to internal peers to
+         indicate the degree of preference for externally learned
+         routes.  The route with the highest local preference
+         value is preferred.";
+      reference
+        "RFC 4271.";
+    }
+
+    leaf default-shutdown {
+      type boolean;
+      default "false";
+      description
+        "Apply administrative shutdown to newly configured peers.";
+    }
+
+    leaf ebgp-requires-policy {
+      type boolean;
+      default "true";
+      description
+        "When set to 'true' BGP speaker requires in and out policy
+         for EBGP peers (RFC8212).";
+    }
+  }
+
+  grouping global-neighbor-config {
+    container global-neighbor-config {
+      leaf dynamic-neighbors-limit {
+        type uint32 {
+          range "1..5000";
+        }
+        description
+          "Maximum number of BGP Dynamic Neighbors that can be created.";
+      }
+
+      leaf log-neighbor-changes {
+        type boolean;
+        default "false";
+        description
+          "When set to 'true' BGP will start logging neighbor up/down and reset reason.
+           When set to 'false' BGP will stop logging neighbor up/down and reset reason.";
+      }
+
+      container packet-quanta-config {
+        leaf wpkt-quanta {
+          type uint32 {
+            range "1..64";
+          }
+          default "64";
+          description
+            "How many packets to write to peer socket per run.";
+        }
+
+        leaf rpkt-quanta {
+          type uint32 {
+            range "1..10";
+          }
+          default "10";
+          description
+            "How many packets to read from peer socket per I/O cycle.";
+        }
+      }
+    }
+  }
+
+  grouping global-update-group-config {
+    container global-update-group-config {
+      leaf subgroup-pkt-queue-size {
+        type uint32 {
+          range "20..100";
+        }
+        default "40";
+        description
+          "Subgroup packet queue size.";
+      }
+
+      leaf coalesce-time {
+        type uint32 {
+          range "0..4294967295";
+        }
+        units "miliseconds";
+        default "1000";
+        description
+          "Configures the Subgroup coalesce timer.";
+      }
+    }
+  }
+
+  grouping global-static-route-config {
+    leaf import-check {
+      type boolean;
+      default "false";
+      description
+        "When set to 'true' bgp creates entries for network statements
+         if a matching prefix exists in the rib. When set to 'false' bgp
+         creates entries for networks that the router cannot reach.";
+    }
+  }
+
+  grouping neighbor-timers {
+    leaf hold-time {
+      type uint16 {
+        range "0 | 3..65535";
+      }
+      units "seconds";
+      default "180";
```

Comment:
We need to figure out how to mesh this default with the `-F datacenter/traditional` thing.

Reply:
I have added the frr-bgp-datacenter.yang module, to solve this problem. 
This would contains the deviation statement. For one example i have added the deviation for hold-time.
Please have a look. I don't know what parameters will change with what value for data center profile.
Please provide the details, i will add the deviation statements accordingly.

```
 |  |  +--rw hold-time?                uint16 <500>
```